src/instrument/mongoose_instrument.erl

```diff
@@ -274,16 +275,46 @@ set_up_and_register_event(EventName, Labels, Config, Events) ->
             {ExistingLabels, _, _} = maps:next(maps:iterator(HandlerMap)),
             case label_keys(ExistingLabels) of
                 LabelKeys ->
-                    Handlers = do_set_up(EventName, Labels, Config),
-                    Events#{EventName := HandlerMap#{Labels => Handlers}};
+                    handle_metrics_config(EventName, Labels, Config, HandlerMap, Events);
                 ExistingKeys ->
                     {error, #{what => inconsistent_labels,
                               event_name => EventName, labels => Labels,
                               existing_label_keys => ExistingKeys}}
             end;
         #{} ->
+            handle_metrics_config(EventName, Labels, Config, new, Events)
+    end.
+
+-spec handle_metrics_config(event_name(), labels(), config(), handler_map() | new, event_map()) ->
+          event_map() | {error, map()}.
+handle_metrics_config(EventName, Labels, Config, HandlerMap, Events) ->
+    case filter_improper_probe_metrics(Config) of
+        {ok, _} ->
             Handlers = do_set_up(EventName, Labels, Config),
-            Events#{EventName => #{Labels => Handlers}}
+            case HandlerMap of
+                new ->
+                    Events#{EventName => #{Labels => Handlers}};
+                _ ->
+                    Events#{EventName := HandlerMap#{Labels => Handlers}}
+            end;
+        {error, FilteredMetrics} ->
+            {error, #{what => non_gauge_metrics_in_probe,
+                      event_name => EventName, labels => Labels,
+                      improper_metrics => FilteredMetrics}}
+    end.
+
+-spec filter_improper_probe_metrics(config()) -> {ok, #{}} | {error, metrics()}.
+filter_improper_probe_metrics(Config) ->
+    case Config of
+        #{probe := _, metrics := Metrics} ->
+            Pred = fun(K, V) -> is_atom(K) andalso V =/= gauge end,
```

Comment:
what does `is_atom/1` check do?
Or should it be something like `(not is_atom(K) orelse V =/= gauge`?

Reply:
I added this check to see if the probe's metric name was an atom, but I realized it's already checked somewhere else. I’ll remove it here since it’s not needed